### PR TITLE
Delete pipeline resources

### PR DESCRIPTION
**Pull request by CarolynMabbott**
# Changes

Change pipeline resources page so they are able to be deleted 
added test 

https://github.com/tektoncd/dashboard/issues/756

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by AlanGreene** [comment]
We should take this opportunity to move the PipelineResources list to use the new Table component.

Have we verified the other dashboard views are resilient to resources being deleted? In particular I'm thinking of the TaskRun and PipelineRun details views which surface information on PipelineResources. If we're making it easier for people to delete these we need to make sure it doesn't break us.

**Comment by CarolynMabbott** [comment]
If you try and rerun the pipelinerun after a resource has been deleted then the pipelinerun will fail with the message `Couldn't retrieve PipelineResource: couldn't retrieve referenced output PipelineResource "image-source7tpzn": pipelineresource.tekton.dev "image-source7tpzn" not found` 

When you click on the link on the link in details page it takes you to the pipelineresources page where it displays an error of `Error loading PipelineResource Not Found`

**Comment by a-roberts** [comment]
> If you try and rerun the pipelinerun after a resource has been deleted then the pipelinerun will fail with the message `Couldn't retrieve PipelineResource: couldn't retrieve referenced output PipelineResource "image-source7tpzn": pipelineresource.tekton.dev "image-source7tpzn" not found`
> 
> When you click on the link on the link in details page it takes you to the pipelineresources page where it displays an error of `Error loading PipelineResource Not Found`
> 
> Megan is currently doing a PR to change all the lists to the new Table components, she is more than happy to change this to a Table in her PR

It would be useful to display a warning (this would be a feature request) that a PipelineRun's referring to non-existent PipelineResources on the page load (or even via an informer/websocket eventually), I'm good for this particular feature to land separately from the Table component usage - will review now as well, thanks for this @CarolynMabbott

File: src/containers/PipelineResources/PipelineResources.js [hunk 1]
```diff
@@ -13,19 +13,16 @@ limitations under the License.
...
-  StructuredListSkeleton,
-  StructuredListWrapper
+  StructuredListSkeleton
 } from 'carbon-components-react';
-import { getErrorMessage, urls } from '@tektoncd/dashboard-utils';
+import { getErrorMessage } from '@tektoncd/dashboard-utils';
 
+import { PipelineResources as PipelineResourcesList } from '@tektoncd/dashboard-components';
 import { fetchPipelineResources } from '../../actions/pipelineResources';
+import { deletePipelineResource } from '../../api';
```
Comment (a-roberts) [inline src/containers/PipelineResources/PipelineResources.js]: Sidenote, I expected this to be an action but see this is how the current deletePipelineRun code works too and we could do that move separately (and display notifications accordingly, emulating the existing secrets behaviour)

**Comment by a-roberts** [comment]
/lgtm
/approve

https://github.com/tektoncd/dashboard/issues/772 this is where @Megan-Wright's making our table usage consistent **including for PipelineResources** so we'll keep an eye on that for removing the structured list usage here

**Pull request merged**